internal/cluster/transition.go

```diff
@@ -347,6 +353,52 @@ func (th *transitionHandler) PostRefreshCluster(reason string) stateswitch.PostT
 	return ret
 }
 
+func (th *transitionHandler) PostRefreshLogsProgress(progress string) stateswitch.PostTransition {
+	return func(sw stateswitch.StateSwitch, args stateswitch.TransitionArgs) error {
+		sCluster, _ := sw.(*stateCluster)
+		return updateLogsProgress(th.log, th.db, sCluster.cluster, swag.StringValue(sCluster.cluster.Status), progress)
+	}
+}
+
+//check if log collection on cluster level reached timeout
+func (th *transitionHandler) IsLogCollectionTimedOut(sw stateswitch.StateSwitch, args stateswitch.TransitionArgs) (bool, error) {
+	sCluster, ok := sw.(*stateCluster)
+	if !ok {
+		th.log.Error("IsLogCollectionTimedOut incompatible type of StateSwitch")
+		return false, errors.New("Cluster IsLogCollectionTimedOut incompatible type of StateSwitch")
+	}
+
+	// if we are already in timeout, return true
+	if sCluster.cluster.LogsInfo == models.LogsStateTimeout {
+		return true, nil
+	}
+	// if we transitioned to the state before the logs were collected, check the timeout
+	// from the time the state machine entered the state
+	if sCluster.cluster.LogsInfo == models.LogsStateEmpty && time.Time(sCluster.cluster.ControllerLogsStartedAt).IsZero() {
+		return time.Since(time.Time(sCluster.cluster.StatusUpdatedAt)) > th.prepareConfig.LogPendingTimeout, nil
```

Comment:
Wouldn't this fail the installation after 6 minutes?
When we start the installation the  All the above conditions are true.
I think you might get a logs collection timeout before the installation started.

Reply:
In the state machine this condition is tested only while installing and yes, it would timeout the waiting for the logs after 6 minutes (I already increased it to 10) because there can be times were the installation fails before the controller is up to send logs. We would not want to wait forever in this case. Yet, we need to wait reasonable time to account for the case when the cluster moves to error but the controller is not yet fully started. It just wait a bit to see that we really do not have controller logs

Reply:
I think you can deduct if you should have controller logs by the bootstrap node status.
If the bootstrap node didn't get to reboot you will not get the controller logs.

Reply:
Yet the controller is up way before that sometimes sending logs + the e2e tests themselves wait for the controller logs beforehand. Doing what you suggested, I agree, will be most predictable but then we will not be able to use this feature in the CI. What do you think it's best? adding the verification or using fixed timeout in the CI? + @tsorya